include/mbgl/shaders/gl/circle.hpp

```diff
@@ -15,56 +15,55 @@ uniform vec2 u_extrude_scale;
 uniform lowp float u_device_pixel_ratio;
 uniform highp float u_camera_to_center_distance;
 
-attribute vec2 a_pos;
-
-varying vec3 v_data;
+layout (location = 0) in vec2 a_pos;
```

Comment:
I'm not sure, but `layout (location = 0)` is mandatory?

Reply:
Coming from desktop GL I'm assuming it is, I can see what happens without layout qualifiers though

Reply:
[Looks like you're probably right](https://registry.khronos.org/OpenGL/specs/es/3.0/GLSL_ES_Specification_3.00.pdf) reading the spec (pg. 40).

Reply:
From what I remember I know that I didn't use it in past project on Android. I know that layout location is needed only if we don't what to call the `glGetVertexAttribLocation`.

Reply:
They're not required, but also not a bad idea.